### PR TITLE
EncapsulateFieldRefactoring - avoid uncompilable code for ReadOnly encapsulation requests

### LINKED ISSUE #5664: EncapsulateField - incorrectly modifies local field references for the Read-Only encapsulation option
Version 2.5.1.33112
OS: Microsoft Windows NT 10.0.18363.0, x64
Host Product: Microsoft Office x64
Host Version: 16.0.13628.20274
Host Executable: EXCEL.EXE

**Description**
When a field is encapsulated as Read-Only, the refactoring modifies local write-references as though a Let/Set property exists.

1. Starting with:
```
Option Explicit

Public mStuff As Collection

Private Sub Class_Initialize()
    Set mStuff = New Collection
End Sub
```
2. Encapsulate `mStuff` specifying encapsulation as a Read-Only property.  The result is:
```
Option Explicit

Private mStuff As Collection

Public Property Get Stuff() As Collection
    Set Stuff = mStuff
End Property

Private Sub Class_Initialize()
    Set Stuff = New Collection
End Sub
```
3. The reference in `Class_Initialize()` is refactored to reference a non-existing Let/Set property rather than `mStuff` (or `this.Stuff` if a `Private Type` is used)

**Expected behavior**
When the user specifies a Read-Only encapsulation, the refactoring should ensure local field 'write' references use the local field (or, if wrapped in a `Private Type` field, _`this.Member`_ ).  The correct result for the above example would be:
```
Option Explicit

Private mStuff As Collection

Public Property Get Stuff() As Collection
    Set Stuff = mStuff
End Property

Private Sub Class_Initialize()
    Set mStuff = New Collection
End Sub
```

### PR DESCRIPTION
**Pull request by BZngr**
Closes #5664

In the process of fixing issue #5664, it became clear that the `IdentifierReference` aspect of EF was under-represented within the existing Unit Tests.  Also, the resolution was more complicated than anticipated by the fact that the existing `IdentifierReference` replacements were handled by two different RefactoringActions - depending on the whether the field was to be wrapped in a Private Type or not.  

This PR consolidates the multiple `IdentifierReference` refactoring actions to a single `EncapsulateFieldReferenceReplacer` class.  And, an expanded set of unit tests focusing solely on `IdentifierReference` modifications has been added.  Pre-existing EF unit tests focusing on `IdentifierReference` modifications have been moved to the `EncapsulateFieldReferenceReplacer` unit tests. 

Other Related changes:

**UI Client:** 
Currently it is possible for the user to select a ReadOnly encapsulation of a Public field that has existing external assignment references.  To manage user selections, this PR modifies the UI to disable the ReadOnly option for those fields.

**RefactoringAction Clients:**
It is possible (one day) that an EF RefactoringAction could be called from another RefactoringAction.  Without having a UI to manage valid options, the EF request could include ReadOnly encapsulations for fields with existing external assignment references.    To avoid generating uncompilable code, the ReadOnly flag value is ignored and all Let/Set/Get properties are generated as needed.

### CODE REVIEW

File: Rubberduck.Main/Root/RubberduckIoCInstaller.cs
Comment (retailcoder) [inline Rubberduck.Main/Root/RubberduckIoCInstaller.cs:412]: Why `.Kernel` here?

**Review by retailcoder — APPROVED** [review]
LGTM, just one question mark about the IoC registration.

File: Rubberduck.Main/Root/RubberduckIoCInstaller.cs
Comment (BZngr) [reply Rubberduck.Main/Root/RubberduckIoCInstaller.cs:412]: No legitimate reason....`container.Kernel.Register` was a copy/paste transfer from how it was registered in `EncapsulateFieldTestsResolver`.  From some quick research, looks like the recommended way is just `container.Register` (as it's done in the IoCInstaller)...wish I had spotted this inconsistency earlier.   I'll get this changed and also update `EncapsulateFieldTestsResolver` to drop the use of `.Kernel` as well.

Comment (retailcoder) [reply Rubberduck.Main/Root/RubberduckIoCInstaller.cs:412]: > wish I had spotted this inconsistency earlier

*this is why we do reviews* 😉

**Comment by MDoerner** [comment]
In light of PR #5864, it would be good if this PR did not remove the `ReplaceReferencesRefactoringAction`.

**Comment by BZngr** [comment]
@MDoerner excellent catch! Thanks.  I'm  thinking that syncing/restoring `ReplaceReferencesRefactoringAction` after #5864 is merged would be the preferred sequence of events.  Hence, the 'Blocked by other issue' label.

**Pull request merged**
